### PR TITLE
erase ValueClasses defined in tasty

### PR DESCRIPTION
**Pull request by bishabosha**
it's not yet possible for scala 2 classes until we read the parents.

For this reason this PR is not enough to fix the issue https://github.com/scalacenter/tasty-query/issues/119, we should read parents for scala 2 classes

### CODE REVIEW

**Comment by bishabosha** [comment]
@sjrd should scala 2 class parents be read in another PR or this one?

**Review by sjrd — CHANGES_REQUESTED** [review]

File: test-sources/src/main/scala/inheritance/MyArrayOps.scala [hunk 1]
```diff
@@ -0,0 +1,6 @@
+package inheritance
+
+class MyArrayOps[T](val value: Array[T]) extends AnyVal
+
+object MyArrayOps:
+  def genericArrayOps[T](xs: Array[T]): MyArrayOps[T] = new MyArrayOps(xs)
```
[the remark this answers is not in the document]
Comment (bishabosha) [reply test-sources/src/main/scala/inheritance/MyArrayOps.scala:7]: ok now I see that the erasure I implemented doesn't actually consider any type parameters

[the remark this answers is not in the document]
Comment (bishabosha) [reply test-sources/src/main/scala/inheritance/MyArrayOps.scala:7]: need to view the constructor argument with any type arguments from the class substituted by type arguments from the applied type. I don't think this is implemented yet.

[the remark this answers is not in the document]
Comment (bishabosha) [reply test-sources/src/main/scala/inheritance/MyArrayOps.scala:7]: I tried `def subst(tp: Type, from: List[Symbol], to: List[Type])(using Context): Type` but that one didn't seem to work so far

[the remark this answers is not in the document]
Comment (bishabosha) [reply test-sources/src/main/scala/inheritance/MyArrayOps.scala:7]: ok actually we have to substitute using the constructors type parameters

File: shared/src/main/scala/tastyquery/ast/Types.scala [hunk 2]
```diff
@@ -538,7 +543,7 @@ object Types {
         val computed = computeName
         myName = computed
         computed
-      else local
+      else local.asInstanceOf[ThisName]
```
Comment (bishabosha) [inline shared/src/main/scala/tastyquery/ast/Types.scala:546]: this one not strictly necessary but it seems to cause the debugger expression evaluator problems if its not there @adpi2 (for example breakpoint inside of `Types.ErasedTypeRef.preErase`)

File: shared/src/main/scala/tastyquery/ast/Types.scala [hunk 3]
```diff
@@ -102,8 +102,17 @@ object Types {
           else
             tycon match
               case tycon: Symbolic if tycon.symbol.isClass =>
-                // Fast path
-                ClassRef(tycon.symbol.asClass)
+                val cls = tycon.symbol.asClass
+                if cls.isDerivedValueClass then
```
Comment (sjrd) [inline shared/src/main/scala/tastyquery/ast/Types.scala:106]: You'll also have to perform this check in the `case tpe: Symbolic` case, for monomorphic value classes.

**Review by sjrd — CHANGES_REQUESTED** [review]

File: shared/src/main/scala/tastyquery/ast/Types.scala (hunk 2 above)
Comment (sjrd) [reply shared/src/main/scala/tastyquery/ast/Types.scala:550]: Add a comment here, so that we don't remove it later for cleaning up?

Comment (adpi2) [reply shared/src/main/scala/tastyquery/ast/Types.scala]: The problem here is that the scalac options are not used by the expression evaluator.

I am currently working on it.

**Review by sjrd — APPROVED** [review]
LGTM other than the `Scala2Constructor` comment.

**Pull request merged**